lib/handler/url_check.go

```diff
@@ -17,44 +16,50 @@ func urlCheck(session *discordgo.Session, orgMsg *discordgo.MessageCreate) {
-			session.ChannelMessageSendReply(orgMsg.ChannelID, config.CurrentConfig.Duplicate.Message+"\nhttps://discord.com/channels/"+orgMsg.GuildID+"/"+channelid+"/"+messageid, orgMsg.Reference())
+			go session.MessageReactionAdd(orgMsg.ChannelID, orgMsg.ID, config.CurrentConfig.Duplicate.React)
+			replyMessage := config.CurrentConfig.Duplicate.Message + "\nhttps://discord.com/channels/" + orgMsg.GuildID + "/" + channelid + "/" + messageid
+			msg, _ := session.ChannelMessageSendReply(orgMsg.ChannelID, replyMessage, orgMsg.Reference())
+			go session.MessageReactionAdd(msg.ChannelID, msg.ID, config.CurrentConfig.Duplicate.Delete)
+			session.AddHandler(func(s *discordgo.Session, r *discordgo.MessageReactionAdd) {
+				if r.MessageID == msg.ID && r.UserID != s.State.User.ID {
+					go s.MessageReactionRemove(msg.ChannelID, orgMsg.ID, config.CurrentConfig.Duplicate.React, s.State.User.ID)
+					go s.MessageReactionsRemoveAll(msg.ChannelID, msg.ID)
+					go s.ChannelMessageEdit(msg.ChannelID, msg.ID, config.CurrentConfig.Duplicate.Bye)
+					go func() {
+						<-time.After(3 * time.Second)
+						s.ChannelMessageDelete(msg.ChannelID, msg.ID)
+					}()
+				}
+			})
+
 		} else {
 			db.AddLog(orgMsg, &orgMsg.GuildID, &url, &orgMsg.ChannelID, &orgMsg.ID)
 		}
 	}
 }
 
 var (
-	urlWithPathRegex = regexp.MustCompile(`https?://[\w+.:?#[\]@!$&'()~*,;=/%-]+`)
+	re = regexp.MustCompile(`https?://[\w+.:?#[\]@!$&'()~*,;=/%-]+`)
 )
 
 func parseMsg(origin string) []string {
-	result := []string{}
-	if strings.HasPrefix(origin, "< ") {
-		return result
-	}
-url_loop:
-	for _, rawUrl := range urlWithPathRegex.FindAllString(origin, -1) {
-		rawUrl = strings.ToLower(rawUrl)
-		rawUrl = strings.ReplaceAll(rawUrl, "youtu.be/", "youtube.com/watch?v=")
-		Url, err := url.Parse(rawUrl)
-		if err != nil {
-			log.Panic("Invalid URL: ", rawUrl)
-		}
-		for _, v := range config.CurrentConfig.DomainBlacklist {
-			if strings.Contains(Url.Host, v) {
-				continue url_loop
-			}
+	results := re.FindAllString(origin, -1)
+
+	for i, result := range results {
+		results[i] = strings.ReplaceAll(result, "www.", "")
+
+		if strings.HasSuffix(result, "/") {
+			results[i] = result[:len(result)-1]
 		}
-		Url.Host = strings.TrimPrefix(Url.Host, "www.")
-		Url.Path = strings.TrimSuffix(Url.Path, "/")
-		if Url.Host == "twitter.com" {
-			Url.Host = strings.TrimPrefix(Url.Host, "m.")
-			Url.Host = strings.TrimPrefix(Url.Host, "mobile.")
-			Url.RawQuery = ""
+
+		if strings.Contains(result, "youtu.be") || strings.Contains(result, ".mobile") {
+			results[i] = strings.ReplaceAll(results[i], "youtu.be/", "youtube.com/watch?v=")
+			results[i] = strings.ReplaceAll(results[i], ".mobile", "")
+		} else if strings.Contains(result, "twitter.com") && (strings.Contains(result, "m.") || strings.Contains(result, "mobile.")) {
+			results[i] = strings.ReplaceAll(results[i], "m.", "")
+			results[i] = strings.ReplaceAll(results[i], "mobile.", "")
```

Comment:
Why do you explicitly check Contains before ReplaceAll?
If it don't contains, ReplaceAll will do nothing.

Reply:
OK. I understand